includes/class-amp-story-post-type.php

```diff
@@ -1270,6 +1287,122 @@ public static function render_block_with_google_fonts( $block_content, $block )
 		return $block_content;
 	}
 
+	/**
+	 * Converts pixel to percentage based on the editor page sizes.
+	 *
+	 * @param string $axis Axis: x or y.
+	 * @param number $pixels Pixel value.
+	 *
+	 * @return int|string Percentage value compared to AMP Story editor page.
+	 */
+	public static function get_percentage_from_pixels( $axis, $pixels ) {
+		if ( 'x' === $axis ) {
+			return number_format( ( ( $pixels / self::STORY_PAGE_INNER_WIDTH ) * 100 ), 2 );
+		} elseif ( 'y' === $axis ) {
+			return number_format( ( ( $pixels / self::STORY_PAGE_INNER_HEIGHT ) * 100 ), 2 );
+		}
+		return 0;
+	}
+
+	/**
+	 * Wraps each movable block into amp-story-grid-layer and animation wrapper with necessary attributes.
+	 *
+	 * @param string $block_content The block content about to be appended.
+	 * @param array  $block         The full block, including name and attributes.
+	 *
+	 * @return string Modified content.
+	 */
+	public static function render_block_with_grid_layer( $block_content, $block ) {
+
+		$post = get_post();
+		if ( ! $post || self::POST_TYPE_SLUG !== $post->post_type ) {
+			return $block_content;
+		}
+
+		// If the block content already includes amp-story-grid-layer, stop.
+		if ( false !== strpos( $block_content, 'amp-story-grid-layer' ) ) {
+			return $block_content;
+		}
+
+		$movable_blocks = [
+			'core/code',
+			'core/embed',
+			'core/image',
+			'core/list',
+			'core/preformatted',
+			'core/pullquote',
+			'core/quote',
+			'core/table',
+			'core/verse',
+			'core/video',
+			'amp/amp-story-text',
+			'amp/amp-story-post-author',
+			'amp/amp-story-post-date',
+			'amp/amp-story-post-title',
+			'core/html',
+			'core/block', // Reusable blocks.
+			'core/template', // Reusable blocks.
+		];
+
+		$name = $block['blockName'];
+
+		// If the block is not movable, it doesn't need the wrapper.
+		if ( ! in_array( $name, $movable_blocks, true ) ) {
+			return $block_content;
+		}
+
+		$atts         = $block['attrs'];
+		$wrapper_atts = [];
+
+		$style = [
+			'position' => 'absolute',
+		];
+
+		$style['top']  = empty( $atts['positionTop'] ) ? '0%' : $atts['positionTop'] . '%';
+		$style['left'] = empty( $atts['positionLeft'] ) ? '5%' : $atts['positionLeft'] . '%';
+		if ( isset( $atts['width'] ) ) {
+			$style['width'] = self::get_percentage_from_pixels( 'x', $atts['width'] ) . '%';
+		}
+		if ( isset( $atts['height'] ) ) {
+			$style['height'] = self::get_percentage_from_pixels( 'y', $atts['height'] ) . '%';
+		}
+
+		$wrapper_style = isset( $atts['style'] ) ? $atts['style'] : '';
+
+		foreach ( $style as $att => $value ) {
+			$wrapper_style .= "$att:$value;";
+		}
+
+		$wrapper_atts['style'] = $wrapper_style;
```

Comment:
Should this be conditional based on whether `! empty( $wrapper_atts['style'] )`?